bats/tests/profile/create-profile-output.bats

```diff
@@ -0,0 +1,284 @@
+    assert_output --partial - <<'EOF'
+Windows Registry Editor Version 5.00
+[HKEY_CURRENT_USER\SOFTWARE\Policies]
+[HKEY_CURRENT_USER\SOFTWARE\Policies\Rancher Desktop]
+[HKEY_CURRENT_USER\SOFTWARE\Policies\Rancher Desktop\defaults]
+[HKEY_CURRENT_USER\SOFTWARE\Policies\Rancher Desktop\defaults\containerEngine]
+[HKEY_CURRENT_USER\SOFTWARE\Policies\Rancher Desktop\defaults\containerEngine\allowedImages]
+"patterns"=hex(7):61,00,62,00,63,00,00,00,67,00,68,00,69,00,00,00,64,00,65,00,66,00,00,00,00,00
+[HKEY_CURRENT_USER\SOFTWARE\Policies\Rancher Desktop\defaults\kubernetes]
+"enabled"=dword:0
+[HKEY_CURRENT_USER\SOFTWARE\Policies\Rancher Desktop\defaults\WSL]
+[HKEY_CURRENT_USER\SOFTWARE\Policies\Rancher Desktop\defaults\WSL\integrations]
+"first"=dword:1
+"second"=dword:0
+EOF
+}
+
+@test 'encodes multi-string values and maps from a file' {
+    run rdctl create-profile --output reg --hive hkcu --input <(complex_json_data)
+    assert_registry_output
+}
+
+@test 'encodes multi-string values and maps from a json string' {
+    run rdctl create-profile --output reg --hive hkcu --body "$(complex_json_data)"
+    assert_registry_output
+}
+
+@test 'complains when no input source is specified' {
+    for type in reg plist; do
+        run rdctl create-profile --output $type
+        assert_failure
+        assert_output --partial "no input format specified: must specify exactly one input format of '--input FILE|-', '--body|-b STRING', or '--from-settings'"
+    done
+}
+
+@test 'complains when multiple input sources are specified' {
+    for type in reg plist; do
+        run rdctl create-profile --output $type --input some-file.txt -b moose
+        assert_failure
+        assert_output --partial "too many input format specified: must specify exactly one input format of '--input FILE|-', '--body|-b STRING', or '--from-settings'"
+
+        run rdctl create-profile --output $type --input some-file.txt --from-settings
+        assert_failure
+        assert_output --partial "too many input format specified: must specify exactly one input format of '--input FILE|-', '--body|-b STRING', or '--from-settings'"
+
+        run rdctl create-profile --output $type --input some-file.txt -b moose --from-settings
+        assert_failure
+        assert_output --partial "too many input format specified: must specify exactly one input format of '--input FILE|-', '--body|-b STRING', or '--from-settings'"
+
+        run rdctl create-profile --output $type -b moose --from-settings
+        assert_failure
+        assert_output --partial "too many input format specified: must specify exactly one input format of '--input FILE|-', '--body|-b STRING', or '--from-settings'"
+
+    done
+}
+
+simple_json_data() {
+    echo '{ "kubernetes": {"version": "moose-head" }}'
+}
+
+assert_moose_head_plist_output() {
+    assert_success
+    # Just match a few of the lines near the start and the end of the output.
+    # The unit tests do more comprehensive output checking.
```

Comment:
The comment seems out-dated.

Reply:
Yes, I'm going with full here-doc testing now.